nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+/* *****************************************************************************
+ * Copyright (c) 2015-2018 Konduit k.k.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.linalg.factory.ops;
+
+import org.junit.Test;
+import org.nd4j.linalg.BaseNd4jTest;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.factory.Nd4jBackend;
+import org.nd4j.linalg.indexing.conditions.Conditions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class NDBaseTest extends BaseNd4jTest {
+    public NDBaseTest(Nd4jBackend backend) {
+        super(backend);
+    }
+
+    @Test
+    public void testAll() {
+        NDBase base = new NDBase();
```

Comment:
Hm... NDBase is a special case.
On the one hand we don't want this (i.e., don't want `new NDBase()` in the tests) on the other hand it's not going to be as simple as adding it like the others - https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java#L121-L164

I'll need to look at that. Just FYI we'll remove the `new NDBase()` at some point